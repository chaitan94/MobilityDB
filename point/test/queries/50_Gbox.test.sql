/*****************************************************************************
 * Gbox
 *****************************************************************************/

SELECT gbox 'GBOX((1.0, 2.0), (1.0, 2.0))';
SELECT gbox 'GBOX Z((1.0, 2.0, 3.0), (1.0, 2.0, 3.0))';
SELECT gbox 'GBOX M((1.0, 2.0, 3.0), (1.0, 2.0, 3.0))';
SELECT gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))';
SELECT gbox 'GEODBOX((1.0, 2.0, 3.0), (1.0, 2.0, 3.0))';
SELECT gbox 'GEODBOX M((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))';

SELECT gbox(1,2,3,4);
SELECT gbox(1,2,3,4,5,6);
SELECT gbox(1,2,3,4,5,6,7,8);
SELECT gbox3dm(1,2,3,4,5,6);
SELECT geodbox(1,2,3,4,5,6);
SELECT geodbox(1,2,3,4,5,6,7,8);

SELECT gbox(8,7,6,5,4,3,2,1);
SELECT gbox3dm(6,5,4,3,2,1);
SELECT geodbox(8,7,6,5,4,3,2,1);

SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((2.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))'); 
SELECT gbox_cmp(gbox 'GBOX ZM((2.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))'); 
SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 3.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))'); 
SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 3.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))'); 
SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 4.0, 4.0), (1.0, 2.0, 3.0, 4.0))'); 
SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 4.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))'); 
SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 5.0), (1.0, 2.0, 3.0, 4.0))'); 
SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 5.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))'); 
SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (2.0, 2.0, 3.0, 4.0))'); 
SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (2.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))'); 
SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 3.0, 3.0, 4.0))'); 
SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 3.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))'); 
SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 4.0, 4.0))'); 
SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 4.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))'); 
SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 5.0))'); 
SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 5.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))'); 

SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))'); 

-------------------------------------------------------------------------------

SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b = t2.b;
SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b <> t2.b;
SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b < t2.b;
SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b <= t2.b;
SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b > t2.b;
SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b >= t2.b;

SELECT count(*) FROM tbl_tgeompoint WHERE temp::gbox IS NOT NULL;
SELECT count(*) FROM tbl_tgeogpoint WHERE temp::gbox IS NOT NULL;

SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b && t2.b;
SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b @> t2.b;
SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b <@ t2.b;
SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b ~= t2.b;

/*****************************************************************************/