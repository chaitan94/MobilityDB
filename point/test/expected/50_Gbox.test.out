/*****************************************************************************
 * Gbox
 *****************************************************************************/
SELECT gbox 'GBOX((1.0, 2.0), (1.0, 2.0))';
       gbox        
-------------------
 GBOX((1,2),(1,2))
(1 row)

SELECT gbox 'GBOX Z((1.0, 2.0, 3.0), (1.0, 2.0, 3.0))';
          gbox           
-------------------------
 GBOX Z((1,2,3),(1,2,3))
(1 row)

SELECT gbox 'GBOX M((1.0, 2.0, 3.0), (1.0, 2.0, 3.0))';
          gbox           
-------------------------
 GBOX M((1,2,3),(1,2,3))
(1 row)

SELECT gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))';
             gbox             
------------------------------
 GBOX ZM((1,2,3,4),(1,2,3,4))
(1 row)

SELECT gbox 'GEODBOX((1.0, 2.0, 3.0), (1.0, 2.0, 3.0))';
           gbox           
--------------------------
 GEODBOX((1,2,3),(1,2,3))
(1 row)

SELECT gbox 'GEODBOX M((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))';
              gbox              
--------------------------------
 GEODBOX M((1,2,3,4),(1,2,3,4))
(1 row)

/* Errors */
SELECT gbox 'AAA(1, 2, 3)';
ERROR:  Could not parse GBOX
LINE 2: SELECT gbox 'AAA(1, 2, 3)';
                    ^
SELECT gbox 'gbox(1, 2, 3)';
ERROR:  Could not parse GBOX
LINE 1: SELECT gbox 'gbox(1, 2, 3)';
                    ^
SELECT gbox 'gbox((AA, 2, 3))';
ERROR:  Could not parse GBOX
LINE 1: SELECT gbox 'gbox((AA, 2, 3))';
                    ^
SELECT gbox 'gbox((1, AA, 3))';
ERROR:  Could not parse GBOX
LINE 1: SELECT gbox 'gbox((1, AA, 3))';
                    ^
SELECT gbox 'gbox z((1, 2, AA))';
ERROR:  Could not parse GBOX
LINE 1: SELECT gbox 'gbox z((1, 2, AA))';
                    ^
SELECT gbox 'gbox m((1, 2, AA))';
ERROR:  Could not parse GBOX
LINE 1: SELECT gbox 'gbox m((1, 2, AA))';
                    ^
SELECT gbox 'gbox((1, 2, 3))';
ERROR:  Could not parse GBOX
LINE 1: SELECT gbox 'gbox((1, 2, 3))';
                    ^
SELECT gbox 'gbox m((1, 2, 3))';
ERROR:  Could not parse GBOX
LINE 1: SELECT gbox 'gbox m((1, 2, 3))';
                    ^
SELECT gbox 'gbox m((1, 2, 3),()';
ERROR:  Could not parse GBOX
LINE 1: SELECT gbox 'gbox m((1, 2, 3),()';
                    ^
SELECT gbox 'gbox m((1, 2, 3),(1)';
ERROR:  Could not parse GBOX
LINE 1: SELECT gbox 'gbox m((1, 2, 3),(1)';
                    ^
SELECT gbox 'gbox z((1, 2, 3),(1,2)';
ERROR:  Could not parse GBOX
LINE 1: SELECT gbox 'gbox z((1, 2, 3),(1,2)';
                    ^
SELECT gbox 'gbox m((1, 2, 3),(1,2)';
ERROR:  Could not parse GBOX
LINE 1: SELECT gbox 'gbox m((1, 2, 3),(1,2)';
                    ^
SELECT gbox 'gbox m((1, 2, 3),(1,2,3)';
ERROR:  Could not parse GBOX
LINE 1: SELECT gbox 'gbox m((1, 2, 3),(1,2,3)';
                    ^
SELECT gbox(1,2,3,4);
       gbox        
-------------------
 GBOX((1,3),(2,4))
(1 row)

SELECT gbox(1,2,3,4,5,6);
          gbox           
-------------------------
 GBOX Z((1,3,5),(2,4,6))
(1 row)

SELECT gbox(1,2,3,4,5,6,7,8);
             gbox             
------------------------------
 GBOX ZM((1,3,5,7),(2,4,6,8))
(1 row)

SELECT gbox3dm(1,2,3,4,5,6);
         gbox3dm         
-------------------------
 GBOX M((1,3,5),(2,4,6))
(1 row)

SELECT geodbox(1,2,3,4,5,6);
         geodbox          
--------------------------
 GEODBOX((1,3,5),(2,4,6))
(1 row)

SELECT geodbox(1,2,3,4,5,6,7,8);
            geodbox             
--------------------------------
 GEODBOX M((1,3,5,7),(2,4,6,8))
(1 row)

SELECT gbox(8,7,6,5,4,3,2,1);
             gbox             
------------------------------
 GBOX ZM((7,5,3,1),(8,6,4,2))
(1 row)

SELECT gbox3dm(6,5,4,3,2,1);
         gbox3dm         
-------------------------
 GBOX M((5,3,1),(6,4,2))
(1 row)

SELECT geodbox(8,7,6,5,4,3,2,1);
            geodbox             
--------------------------------
 GEODBOX M((7,5,3,1),(8,6,4,2))
(1 row)

SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((2.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))');
 gbox_cmp 
----------
       -1
(1 row)

SELECT gbox_cmp(gbox 'GBOX ZM((2.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))');
 gbox_cmp 
----------
        1
(1 row)

SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 3.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))');
 gbox_cmp 
----------
       -1
(1 row)

SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 3.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))');
 gbox_cmp 
----------
        1
(1 row)

SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 4.0, 4.0), (1.0, 2.0, 3.0, 4.0))');
 gbox_cmp 
----------
       -1
(1 row)

SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 4.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))');
 gbox_cmp 
----------
        1
(1 row)

SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 5.0), (1.0, 2.0, 3.0, 4.0))');
 gbox_cmp 
----------
       -1
(1 row)

SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 5.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))');
 gbox_cmp 
----------
        1
(1 row)

SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (2.0, 2.0, 3.0, 4.0))');
 gbox_cmp 
----------
       -1
(1 row)

SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (2.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))');
 gbox_cmp 
----------
        1
(1 row)

SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 3.0, 3.0, 4.0))');
 gbox_cmp 
----------
       -1
(1 row)

SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 3.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))');
 gbox_cmp 
----------
        1
(1 row)

SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 4.0, 4.0))');
 gbox_cmp 
----------
       -1
(1 row)

SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 4.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))');
 gbox_cmp 
----------
        1
(1 row)

SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 5.0))');
 gbox_cmp 
----------
       -1
(1 row)

SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 5.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))');
 gbox_cmp 
----------
        1
(1 row)

SELECT gbox_cmp(gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))', gbox 'GBOX ZM((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))');
 gbox_cmp 
----------
        0
(1 row)

SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b = t2.b;
 count 
-------
   100
(1 row)

SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b <> t2.b;
 count 
-------
  9900
(1 row)

SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b < t2.b;
 count 
-------
  4950
(1 row)

SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b <= t2.b;
 count 
-------
  5050
(1 row)

SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b > t2.b;
 count 
-------
  4950
(1 row)

SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b >= t2.b;
 count 
-------
  5050
(1 row)

SELECT count(*) FROM tbl_tgeompoint WHERE temp::gbox IS NOT NULL;
 count 
-------
   100
(1 row)

SELECT count(*) FROM tbl_tgeogpoint WHERE temp::gbox IS NOT NULL;
 count 
-------
   100
(1 row)

SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b && t2.b;
 count 
-------
   100
(1 row)

SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b @> t2.b;
 count 
-------
   100
(1 row)

SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b <@ t2.b;
 count 
-------
   100
(1 row)

SELECT count(*) FROM tbl_gbox t1, tbl_gbox t2 where t1.b ~= t2.b;
 count 
-------
   100
(1 row)

/*****************************************************************************/
